openpype/hosts/maya/api/pipeline.py

```diff
@@ -681,6 +685,105 @@ def before_workfile_save(event):
         create_workspace_mel(workdir_path, project_name)
 
 
+def display_warning(message, show_cancel=False):
+    """Show feedback to user.
+
+        Returns:
+            bool
+        """
+
+    from qtpy import QtWidgets
+
+    accept = QtWidgets.QMessageBox.Ok
+    if show_cancel:
+        buttons = accept | QtWidgets.QMessageBox.Cancel
+    else:
+        buttons = accept
+
+    state = QtWidgets.QMessageBox.warning(
+        None,
+        "",
+        message,
+        buttons=buttons,
+        defaultButton=accept
+    )
+
+    return state == accept
+
+
+def workfile_save_before_xgen(event):
```

Comment:
Can we add a docstring as to what this does and why it's needed?

Reply:
👍